client/js/editor/viewport.js

```diff
@@ -330,6 +338,24 @@ shapy.editor.Viewport = function(name) {
    * @public {!shapy.editor.Viewport.Type}
    */
   this.type = shapy.editor.Viewport.Type.PERSPECTIVE;
+
+  /**
+   * Current position of the mouse.
+   * @private {!goog.vec.Vec2.Float32}
+   */
+  this.currMousePos = goog.vec.Vec2.createFloat32();
```

Comment:
I think it's better to use goog.math.Vec2 for 2D coordinates since you can refer to the two elements as vec.x and vec.y instead of vec[0] and vec[1], which is much easier to read and understand. The main reason for using goog.vec.Vec3/4 is that those structures are much easier to pass to WebGL and are easier to multiply/normalize etc.
But for 2D stuff, we do not really care about performance since we do not do that many operations, but we do care about the code being readable since the logic might get quite complicated.